core/src/main/scala/kafka/coordinator/group/GroupCoordinatorAdapter.scala

```diff
@@ -397,19 +389,24 @@ class GroupCoordinatorAdapter(
     request: TxnOffsetCommitRequestData,
     bufferSupplier: BufferSupplier
   ): CompletableFuture[TxnOffsetCommitResponseData] = {
+    val currentTimeMs = time.milliseconds
     val future = new CompletableFuture[TxnOffsetCommitResponseData]()
 
     def callback(results: Map[TopicPartition, Errors]): Unit = {
       val response = new TxnOffsetCommitResponseData()
-      val byTopics = new util.HashMap[String, TxnOffsetCommitResponseData.TxnOffsetCommitResponseTopic]()
+      val byTopics = new mutable.HashMap[String, TxnOffsetCommitResponseData.TxnOffsetCommitResponseTopic]()
 
       results.forKeyValue { (tp, error) =>
-        var topic = byTopics.get(tp.topic)
-        if (topic == null) {
-          topic = new TxnOffsetCommitResponseData.TxnOffsetCommitResponseTopic().setName(tp.topic)
-          byTopics.put(tp.topic, topic)
-          response.topics.add(topic)
+        val topic = byTopics.get(tp.topic) match {
```

Comment:
This seems like it could use the getOrElseUpdate method, but I'm not sure if we typically have side effects like adding to the response. Not a huge deal either way.

Reply:
Yeah, I didn’t use it because of the side effect.

Reply:
The downside is double-lookup (one for get and another for insert).